packages/remix-react/transition.ts

```diff
@@ -2077,11 +2081,17 @@ async function makeDeferredLoaderData(
               },
             };
 
-            if (signal!.aborted) return;
             update({
               deferredLoaderData: newDeferredLoaderData,
               loaderData: newLoaderData,
             });
+
+            if (
+              Object.keys(newDeferredLoaderData[match.route.id]).length === 0
+            ) {
+              // Mark this deferred as no longer pending once we've updated state
+              pendingNavigationDeferredControllers.delete(match.route.id);
```

Comment:
Once a deferred has completely resolved - and we have no pending promises left (and therefore nothing to abort) we can remove it from `pendingNavigationDeferredControllers` so it doesn't think it got cancelled on subsequent actions or navigations